src/OpenTelemetry/Logs/Options/OpenTelemetryLoggerOptions.cs

```diff
@@ -29,29 +30,57 @@ public class OpenTelemetryLoggerOptions
         internal readonly List<BaseProcessor<LogRecord>> Processors = new();
         internal ResourceBuilder ResourceBuilder = ResourceBuilder.CreateDefault();
 
+        /// <summary>
+        /// Gets or sets a value indicating whether or not formatted log message
+        /// should be included on generated <see cref="LogRecord"/>s. Default
+        /// value: <see langword="false"/>.
+        /// </summary>
+        /// <remarks>
+        /// Note: When set to <see langword="false"/> a formatted log message
+        /// will not be included if a message template can be found. If a
+        /// message template is not found, a formatted log message is always
+        /// included.
+        /// </remarks>
+        public bool IncludeFormattedMessage { get; set; }
+
         /// <summary>
         /// Gets or sets a value indicating whether or not log scopes should be
         /// included on generated <see cref="LogRecord"/>s. Default value:
-        /// False.
+        /// <see langword="false"/>.
         /// </summary>
         public bool IncludeScopes { get; set; }
 
         /// <summary>
-        /// Gets or sets a value indicating whether or not formatted log message
-        /// should be included on generated <see cref="LogRecord"/>s. Default
-        /// value: False.
+        /// Gets or sets a value indicating whether or not log state should
+        /// be included on generated <see cref="LogRecord"/>s. Default value:
+        /// <see langword="true"/>.
         /// </summary>
-        public bool IncludeFormattedMessage { get; set; }
+        public bool IncludeState { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether or not the <see
+        /// cref="Activity.TraceStateString"/> for the current <see
+        /// cref="Activity"/> should be included on generated <see
+        /// cref="LogRecord"/>s. Default value: <see langword="false"/>.
+        /// </summary>
+        public bool IncludeTraceState { get; set; }
```

Comment:
Not excited about this new option. Is the cost of keeping things as is pretty high? In a lot of cases I think TraceState is null or empty anyways.

Reply:
I'm not passionate that we absolutely need this. What the [Activity.TraceStateString getter](https://github.com/dotnet/runtime/blob/669e5a9e3febe36247ba859aa2e3b828c8a9ed0e/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L773-L782) does is walk the chain and look for a value. So if you happen to be logging inside of a deeply nested span those cycles get spent even if trace state isn't being used. The idea here was to not spend any time on it unless it was needed.

Reply:
I think we should not add this option unless we get perf complaints for this as the use-case it's trying to optimize seems like a rare occurrence.

Reply:
Is it rare though? Using TraceState is probably rare. But whether or not it is used doesn't change the perf hit we take to access it! We have seen some epic deep traces. Let's say you are in an AspNetCore app. Incoming request creates a root span. Then maybe some span for controller. Then maybe some span for service. Then outgoing Http span. If you log at that 4th level trying to read TraceStateString has to look at all 4 of those spans to see if any have TraceState. Just to drop it or not find it at all. If you are writing a lot of logs, it adds up.

I think of this change as more of us moving into spec compliance with logs. [TraceState is not on log data model](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#log-and-event-record-definition). Adding it on `LogRecord` was probably a mistake?

Again I'm not passionate that we need the setting. But I think dropping it by default is good. The setting is to make sure anyone that is actually reliant on the existing behavior can avoid a break.

Alternatively we could obsolete it and tackle this later, but I really don't like the wasted perf 👊

Reply:
Appreciate the perf concerns and trying to remedy the fact that we shipped stable logs before the spec was ready, but I vote this moves to a separate PR for discussion. I'd actually prefer we discuss removing tracestate altogether and just deal with any fallout from this breaking change.